application/app/models/project.rb

```diff
@@ -43,6 +43,18 @@ def files
       end
   end
 
+  def upload_files
+    @upload_files ||=
+      begin
+        directory = File.join(self.class.upload_files_directory(id))
+        Dir.glob(File.join(directory, '*.yml'))
+           .select { |f| File.file?(f) }
+           .sort_by { |f| File.ctime(f) }
+           .map { |f| UploadFile.find(id, File.basename(f, ".yml")) }
+           .compact
+      end
+  end
+
```

Comment:
I think we are missing the UploadCollection / UploadBundle / UploadRepository class here.
A project will return a list of this ^^ object and inside the object there will be a list of files.
The object will be created based on the directories under `project > uploads`

Didn't we agree on these? or am I confused now?

As well, we need to rename project > files to project > downloads

Reply:
will implement that in the next PR